modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/ConnectionProviderAwareMetadataContext.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.metadata;
+
+import org.mule.runtime.api.connection.ConnectionProvider;
+import org.mule.runtime.api.metadata.MetadataContext;
+
+import java.util.Optional;
+
+/**
+ * A {@link MetadataContext} which is aware of the {@link ConnectionProvider} used to provide its connection.
+ *
+ * @since 4.4.0, 4.3.1
```

Comment:
why 4.3.1?

Reply:
This should be cherry-picked to solve this issue for 4.3.x versions. If we decide that that is not necessary, we can add it only in 4.4.0.

Reply:
the bug cannot be reproduced without OCS, which will only be available with 4.4.0

Reply:
👍